apiserver/facades/controller/agenttools/agenttools.go

```diff
@@ -63,13 +66,16 @@ type newEnvironFunc func() (environs.Environ, error)
 type toolsFinder func(context.Context, tools.SimplestreamsFetcher, environs.BootstrapEnviron, int, int, []string, coretools.Filter) (coretools.List, error)
 type envVersionUpdater func(*state.Model, version.Number) error
 
-func checkToolsAvailability(ctx context.Context, newEnviron newEnvironFunc, modelCfg *config.Config, finder toolsFinder) (version.Number, error) {
-	currentVersion, ok := modelCfg.AgentVersion()
-	if !ok || currentVersion == version.Zero {
+func (api *AgentToolsAPI) checkToolsAvailability(ctx context.Context) (version.Number, error) {
+	currentVersion, err := api.modelAgentService.GetModelAgentVersion(ctx)
+	if err != nil {
+		return version.Zero, errors.Annotate(err, "getting agent version from service")
+	}
+	if currentVersion == version.Zero {
```

Comment:
I wonder why we checked for version.Zero. The model agent version is either known or it's an error. I think the new agent service should never return a Zero version without an error, in which case we don't need this check.